src/input/MouseInputSource.js

```diff
@@ -0,0 +1,111 @@
+import InputSource from "./InputSource.js";
+
+/**
+ *  MouseInputSource watches browser MouseEvents and tracks state
+ */
+export default class MouseInputSource extends InputSource {
+  constructor(targetElement = document) {
+    super();
+    this._updatePosition = this._updatePosition.bind(this);
+
+    this._target = null;
+
+    /** X, and Y normalized to [-1, 1] */
+    this._normalizedX = null;
+    this._normalizedY = null;
+
+    this._clientX = null;
+    this._clientY = null;
+    this._offsetX = null;
+    this._offsetY = null;
+    this._screenX = null;
+    this._screenY = null;
+    this._movementX = null;
+    this._movementY = null;
+    this._appX = null;
+    this._appY = null;
+
+    this._primaryButton = false;
```

Comment:
I'd use an array of button states for that instead. As many mouses has more than just 3 buttons (Mine has 11 actually).
And it could simplify the rest of the code with mouseup/mousedown (`buttonStates[ev.button] = true`).
You could add an alias if you feel so to  `/mouse/button/primary` but it should be ok also to do `/mouse/button/4`